lib/internal/errors.js

```diff
@@ -146,7 +126,27 @@ const maybeOverridePrepareStackTrace = (globalThis, error, trace) => {
     return MainContextError.prepareStackTrace(error, trace);
   }
 
-  return kNoOverride;
+  // Normal error formatting:
+  //
+  // Error: Message
+  //     at function (file)
+  //     at file
+  let errorString;
+  // Do not use primordials here: we intercept user code here.
+  const { toString } = error;
+  if (typeof toString === 'function' &&
+      toString !== MainContextObjectToString) {
+    errorString = FunctionPrototypeCall(toString, error);
+    if (errorString === '[Object object]' || typeof errorString !== 'string') {
+      errorString = ErrorPrototypeToString(error);
+    }
```

Comment:
I think the `'[Object object]'` check is not necessary since we checked before that the function is not `%Object.prototype.toString%`. I'm suggesting replacing the `typeof` check with a template string to force it to be a string, and if any error arises, fallback to the primordials. wdyt?
```suggestion
    try {
      errorString = `${FunctionPrototypeCall(toString, error)}`;
    } catch {    
      errorString = ErrorPrototypeToString(error);
    }
```

Reply:
No strong feelings from me, it's hard to find the right balance between helpfulness and "magic" behavior, IMO if the user changes their Error objects to the `toString` method returns e.g. a number, that's on them, and maybe they expect to see a number in place of a stack trace? Or maybe they've made a mistake in their code, and won't understand why their `toString` method is called but its result is not used?

Reply:
I decided to stick to the current behavior in this PR. I do plan on working further on the errors as there is quite a bit room for further improvements later on.